### PR TITLE
Use PinnableSlice in Transactions

### PR DESCRIPTION
**Pull request by maysamyabandeh**
The ::Get from DB is not augmented with an overload method that takes a PinnableSlice instead of a string. Transactions however are not yet upgraded to use the new API. As a result, transaction users such as MyRocks cannot benefit from it. This patch updates the transactional API with a PinnableSlice overload.

### CODE REVIEW

**Comment by maysamyabandeh** [comment]
I got 3% better sysbench throughput with pinnableslice:

$ cat ~/sysbench_results/mysqld.pinslice.1t.1/sb.r.trx.nr1000000.range100.rocksdb.point-query
10635   rocksdb point-query range=100
$ cat ~/sysbench_results/mysqld.pinslice.disabled.1t.1/sb.r.trx.nr1000000.range100.rocksdb.point-query
10339   rocksdb point-query range=100

Since the benefit comes from less memcpy this should also results into saving 1.6% cpu usage of memcpy that we observed in recent perf logs. 

Reproduce steps:
114   ntabs=8
115   nrows=1000000
116   writesecs=60
117   setup=1
118   cleanup=1
119   bench=point-query
120   client=./client.sh

 11 for THREADS in 1 ; do
 12 for i in `seq 1 1`; do
 13
 14 #install_rpm $HOST 3306 $RPM_SRC
 15 #send_sysbench_to_server $HOST
 16
 17 EXPR=mysqld.pinslice.disabled.${THREADS}t.$i
 18 rebuilddb $HOST ROCKSDB 3306 $RPM_SRC
 19 config_db $HOST 3306 "binlog_format='row'"
 20 config_db $HOST 3306 "read_only=off"
 21 config_db $HOST 3306 "rocksdb_enable_2pc=1"
 22 config_db $HOST 3306 "sync_binlog=20"
 23 config_db $HOST 3306 "rocksdb_flush_log_at_trx_commit=2"
 24 config_db $HOST 3306 "binlog_order_commits=1"
 25 run_sysbench $HOST rocksdb $EXPR $DURATION $THREADS
 26 grab_results $HOST
 27 cat $LOCAL_RES_DIR/$EXPR/sb.r.trx*
 28 ~/notify.sh "experiment done"
 29
 30 done #threads
 31 done #i

**Comment by IslamAbdelRahman** [comment]
@maysamyabandeh, can you please look at the test failures ?

**Comment by maysamyabandeh** [comment]
Weird! The tests are passing in my local branch. Let me dig in...

**Comment by maysamyabandeh** [comment]
@IslamAbdelRahman all good now.

File: include/rocksdb/utilities/transaction.h [hunk 1]
```diff
@@ -212,6 +230,23 @@ class Transaction {
...
+  // For backward compatiblity a default implementation is provided
+  virtual Status GetForUpdate(const ReadOptions& options,
+                              ColumnFamilyHandle* column_family,
+                              const Slice& key, PinnableSlice* pinnable_val,
+                              bool exclusive = true) {
+    if (pinnable_val == nullptr) {
+      std::string* null_str = nullptr;
+      return GetForUpdate(options, key, null_str);
+    } else {
+      assert(pinnable_val != nullptr);
```
Comment (yiwu-arbug) [inline include/rocksdb/utilities/transaction.h]: the assert looks like redundant..

**Pull request closed without merging**